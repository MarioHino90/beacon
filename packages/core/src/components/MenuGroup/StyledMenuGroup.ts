import styled, { css } from "styled-components";
import { getColor, getSpacing } from "@braze/beacon-styling";

import { StyledHeading } from "../Heading";
import { StyledMenuButton } from "../MenuButton";

import { StyledMenuGroupProps } from "./types";

export const StyledMenuGroup = styled.div<StyledMenuGroupProps>`
  width: 100%;

  &:not(:last-of-type) {
    padding-bottom: ${getSpacing("md")};
    margin-bottom: ${getSpacing("md")};
    border-bottom: 1px solid ${getColor("gray", 100)};
  }

  & > ${StyledHeading} {
    padding: ${getSpacing("md")} ${getSpacing("xl")} ${getSpacing("sm")};
    color: ${getColor("gray", 500)};
    text-transform: uppercase;
  }

  ${(props) =>
    props.hasHeading &&
    css`
      &:last-of-type {
        margin-bottom: ${getSpacing("sm")};
      }

      & > ${StyledMenuButton} {
        padding-left: ${getSpacing("xxl")};
      }
    `};
`;
