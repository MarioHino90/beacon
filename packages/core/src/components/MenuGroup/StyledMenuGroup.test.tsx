import React from "react";
import { shallow } from "enzyme";
import faker from "@faker-js/faker";
import theme from "@braze/beacon-theme";

import { StyledMenuGroup } from "./StyledMenuGroup";

faker.seed(1603838448143);

describe("StyledMenuGroup", () => {
  let props;

  beforeEach(() => {
    props = {
      children: faker.random.words(),
      className: faker.random.word(),
      id: faker.random.word(),
      theme,
    };
  });

  it("should match the snapshot", () => {
    const wrapper = shallow(<StyledMenuGroup {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("should match the snapshot when 'hasHeading' is defined", () => {
    const wrapper = shallow(<StyledMenuGroup {...props} hasHeading />);

    expect(wrapper).toMatchSnapshot();
  });
});
