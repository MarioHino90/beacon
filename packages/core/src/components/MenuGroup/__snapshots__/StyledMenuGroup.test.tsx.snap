// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledMenuGroup should match the snapshot 1`] = `
.c0 {
  width: 100%;
}

.c0:not(:last-of-type) {
  padding-bottom: 0.75rem;
  margin-bottom: 0.75rem;
  border-bottom: 1px solid #E9EBED;
}

<div
  className="c0 wireless"
  id="Metal"
>
  Pants system transmitting
</div>
`;

exports[`StyledMenuGroup should match the snapshot when 'hasHeading' is defined 1`] = `
.c0 {
  width: 100%;
}

.c0:not(:last-of-type) {
  padding-bottom: 0.75rem;
  margin-bottom: 0.75rem;
  border-bottom: 1px solid #E9EBED;
}

.c0:last-of-type {
  margin-bottom: 0.5rem;
}

<div
  className="c0 Handmade"
  id="JBOD"
>
  Checking
</div>
`;
